### PR TITLE
Feature/dimension based metadata

### PR DESCRIPTION
**Pull request by j-s-rawat**
### What
US: https://trello.com/c/vSxzIedn/5674-update-dp-cantabular-metadata-exporter-to-support-export-based-on-specific-dimensions

Made changes to kafka message and fixed issues related to csv exporter 

### How to review

Describe the steps required to test the changes.

### Who can review

Anyone

### CODE REVIEW

File: handler/handler.go [hunk 1]
```diff
@@ -384,11 +383,12 @@ func (h *InstanceComplete) UpdateInstance(ctx context.Context, e *event.ExportSt
...
-		DatasetID:  e.DatasetID,
-		Edition:    e.Edition,
-		Version:    e.Version,
-		RowCount:   rowCount,
+		InstanceID:   e.InstanceID,
+		DatasetID:    e.DatasetID,
+		Edition:      e.Edition,
+		Version:      e.Version,
+		RowCount:     rowCount,
+		DimensionsID: e.DimensionsID,
```
Comment (ChrysmOre) [inline handler/handler.go:391]: This should be dimensionIDs not dimensionsID

Comment (j-s-rawat) [reply handler/handler.go:391]: done

File: handler/handler.go [hunk 2]
```diff
@@ -130,7 +130,7 @@ func (h *InstanceComplete) getFilterInfo(ctx context.Context, filterOutputID str
 
 	dimensionNames := make([]string, 0)
 	for _, d := range dimensions {
-		dimensionNames = append(dimensionNames, d.Name)
+		dimensionNames = append(dimensionNames, d.ID)
```
Comment (ChrysmOre) [inline handler/handler.go:133]: Again, are we using names or IDs?

File: features/dp-cantabular-csv-exporter-unpublished.feature [hunk 3]
```diff
@@ -104,7 +104,7 @@ Feature: Cantabular-Csv-Exporter-Unpublished
     And cantabular server is healthy
     And cantabular api extension is healthy
 
-    And the following instance with id "instance-happy-01" is available from dp-dataset-api:
+    And the following instance with id "instance-happy-02" is available from dp-dataset-api:
```
Comment (j-s-rawat) [inline features/dp-cantabular-csv-exporter-unpublished.feature:107]: We already have "instance-happy-02" in one of the component tests in the same package. Should adding unique stubs across the package be the strategy to follow, as its a matter of time that we will add another test and face this issue again

File: handler/handler.go [hunk 4]
```diff
@@ -128,14 +128,14 @@ func (h *InstanceComplete) getFilterInfo(ctx context.Context, filterOutputID str
 
 	dimensions := model.Dimensions
 
-	dimensionNames := make([]string, 0)
+	dimensionIds := make([]string, 0)
 	for _, d := range dimensions {
-		dimensionNames = append(dimensionNames, d.Name)
+		dimensionIds = append(dimensionIds, d.ID)
```
Comment (saminahbab) [inline handler/handler.go:133]: This is different to our need to use names with cantabular isn't it?

**Review by saminahbab — APPROVED** [review]

**Pull request merged**